raiden/network/transport/matrix/client.py

```diff
@@ -780,7 +782,16 @@ def _handle_responses(
                 # number of members changed. Verify validity of room
                 if room_members_count != len(room._members):
                     self._handle_member_join_callback(room)
-                all_messages.append((room, sync_room["timeline"]["events"]))
+                all_messages.append(
+                    (
+                        room,
+                        [
+                            message
+                            for message in sync_room["timeline"]["events"]
+                            if message["type"] == "m.room.message"
```

Comment:
Why would there suddenly be other types? We only add a filter...

Reply:
This has nothing to do with the filter. The timeline section also contains state_event changes which are only important for the room state keeping which is already processes by `room.put_event` and `room.process_state_event`. 
In this line we are gathering the messages for further processing in raiden. The other events can be filtered here.

Reply:
What I mean is other types always have been in there